presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/TestSemiTransactionalHiveMetastore.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.hive.metastore;
+
+import com.google.common.collect.ImmutableList;
+import io.prestosql.plugin.hive.authentication.HiveIdentity;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+
+import static com.google.common.util.concurrent.MoreExecutors.directExecutor;
+import static io.prestosql.plugin.hive.HiveTestUtils.HDFS_ENVIRONMENT;
+import static io.prestosql.testing.TestingConnectorSession.SESSION;
+import static java.lang.Math.abs;
+import static java.util.concurrent.Executors.newFixedThreadPool;
+import static org.testng.Assert.assertTrue;
+
+public class TestSemiTransactionalHiveMetastore
```

Comment:
I don’t think we need this test. It’s fragile (as it relies on sleep / timing) and we already have tests to ensure partition dropping works correctly. We don’t usually test that something happens in parallel.

If we did want to test it, a better way would be to use `CountDownLatch` for the `dropPartition` call, as that ensures that multiple threads hit that point concurrently.